### DISCUSSION TITLE
Feature Request: Edit Page's Structure View should have a preview of contents

### DISCUSSION THREAD

**Discussion by cagejsn** [django-cms/django-cms#7440] [General]
Hi all,

I am writing to share something that came up on my project team at work.
We are using Django CMS to power a site for Onboarding people to a company.

We have a team of 2 UX designers, 2 non-technical (change management) and 1 Dev (me).

The UX Designers and the non-technical people are the ones who are actually editing content in the admin & edit pages.

They brought up that the structure view can get overwhelming for the editor.

I propose that the Structure View not use the Name of the Plugin, but rather something customizable which may of may not include the "name" of the plugin.

**Comment by cagejsn** [comment]
This is the view I am talking about:
<img width="806" alt="image" src="https://user-images.githubusercontent.com/8084722/201798756-70a73947-289f-45db-a0aa-b6e760b34eac.png">

When it appears as a side bar on the right I am not seeing the addition descriptive text.

**Comment by marksweb** [comment]
It's up to the plugins, not the CMS itself what is displayed in the structure view.

Take an example from a site of mine using djangocms-frontend where plugins are fairly identifiable

<img width="402" alt="Screenshot 2022-11-15 at 23 52 42" src="https://user-images.githubusercontent.com/1461191/202049418-961a540a-8b91-4aa4-9715-800d90158fff.png">

**Reply by cagejsn** [reply]
How do you configure the Plugin's Text that is shown? I looked in the docs to try and figure this out, and I was unable to. Maybe I missed something. I am happy to update the documentation if it's not present.

**Reply by fsbraun** [reply]
It's the plugin model's `get_short_description()` method that returns a content summary for the plugin.

**Comment by fsbraun** [comment]
What I'd love to see is that the available space is better used, especially if the side bar fills the whole viewport. The text shown is shortened too early after three words by the djangocms-text-ckeditor plugin imho: 
![](https://user-images.githubusercontent.com/8084722/201798756-70a73947-289f-45db-a0aa-b6e760b34eac.png)

**Comment by cagejsn** [comment]
There should be an entry here for `get_short_description` no?
https://docs.django-cms.org/en/latest/reference/plugins.html#cms.models.pluginmodel.CMSPlugin


perhaps also here?
https://docs.django-cms.org/en/latest/how_to/custom_plugins.html#how-to-create-plugins

**Comment by fsbraun** [comment]
Yes, there should! Would you like to create a Pull Request to add this? I'd really appreciate it!

**Reply by cagejsn** [reply]
Yes I can do that!
